compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/util/IrTypeUtils.kt

```diff
@@ -85,4 +88,9 @@ fun IrType.isPrimitiveArray() = isTypeFromKotlinPackage { it in FQ_NAMES.primiti
 
 fun IrType.getPrimitiveArrayElementType() = (this as? IrSimpleType)?.let {
     (it.classifier.owner as? IrClass)?.fqName?.toUnsafe()?.let { fqn -> FQ_NAMES.arrayClassFqNameToPrimitiveType[fqn] }
-}
\ No newline at end of file
+}
+
+fun IrType.isNonPrimitiveArray(): Boolean {
+    val descriptor = classifierOrNull?.descriptor as? ClassDescriptor ?: return false
+    return KotlinBuiltIns.isNonPrimitiveArray(descriptor)
```

Comment:
Please avoid descriptor usage,
here should be smth like
```
(this.isArray() || this.isNullableArray()) && !this.isPrimitiveArray()
```

Reply:
Done